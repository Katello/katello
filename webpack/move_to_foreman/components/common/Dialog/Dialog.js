import React from 'react';
import PropTypes from 'prop-types';
import { Button, Modal, Icon } from 'patternfly-react';

const Dialog = (props) => {
  const buttons = (props.buttons)
    ? props.buttons
    : (
      <Button
        key="cancel"
        bsStyle="default"
        className="btn-cancel"
        onClick={props.onCancel}
      >
        {props.cancelLabel}
      </Button>
    );
  return (
    <Modal show={props.show}>
      <Modal.Header>
        <Button
          className="close"
          onClick={props.onCancel}
          aria-hidden="true"
          aria-label={props.cancelLabel}
        >
          <Icon type="pf" name="close" />
        </Button>
        <Modal.Title>{props.title}</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        {/* eslint-disable react/no-danger */}
        <p dangerouslySetInnerHTML={props.dangerouslySetInnerHTML}>
          {props.message}
        </p>
        {/* eslint-enable react/no-danger */}
      </Modal.Body>
      <Modal.Footer>
        {buttons}
      </Modal.Footer>
    </Modal>
  );
};

Dialog.propTypes = {
  show: PropTypes.bool.isRequired,
  onCancel: PropTypes.func.isRequired,
  message: PropTypes.string,
  title: PropTypes.string.isRequired,
  cancelLabel: PropTypes.string,
  dangerouslySetInnerHTML: PropTypes.shape({}),
  buttons: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

Dialog.defaultProps = {
  cancelLabel: __('Ok'),
  dangerouslySetInnerHTML: undefined,
  message: undefined,
  buttons: undefined,
};

export default Dialog;
