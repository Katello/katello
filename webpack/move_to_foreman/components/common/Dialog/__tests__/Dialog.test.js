import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Button } from 'patternfly-react';
import Dialog from '../Dialog';

describe('Dialog', () => {
  const props = {
    show: true,
    onCancel: () => {},
    title: 'Message dialog',
  };
  const message = 'Some important message';

  it('renders a message dialog', async () => {
    const dialog = shallow(<Dialog
      {...props}
      {...{ message }}
    />);
    expect(toJson(dialog)).toMatchSnapshot();
  });

  it('enables to override cancel button label', async () => {
    const dialog = shallow(<Dialog
      {...props}
      {...{ message }}
      cancelLabel="Custom cancel"
    />);
    expect(toJson(dialog)).toMatchSnapshot();
  });

  it('enables to set inner html', async () => {
    const dialog = shallow(<Dialog
      {...props}
      dangerouslySetInnerHTML={{ _html: '<b>Custom</b> html content' }}
    />);
    expect(toJson(dialog)).toMatchSnapshot();
  });

  it("passes onCancel to the default OK button's onClick", () => {
    const onCancel = jest.fn();
    const dialog = shallow(<Dialog
      {...props}
      {...{ onCancel }}
    />);
    expect(dialog.find('ModalFooter > Button[bsStyle="default"]').props().onClick).toBe(onCancel);
    expect(dialog.find('ModalHeader > Button').props().onClick).toBe(onCancel);
  });

  it('enables to define custom buttons', async () => {
    const dialog = shallow(<Dialog
      {...props}
      buttons={[
        <Button key="btn1">Button 1</Button>,
        <Button key="btn2">Button 2</Button>,
        <Button key="btn3">Button 3</Button>,
      ]}
    />);
    expect(toJson(dialog)).toMatchSnapshot();
  });
});
