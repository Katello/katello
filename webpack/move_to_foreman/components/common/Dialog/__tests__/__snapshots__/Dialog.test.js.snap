// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog enables to define custom buttons 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <Button
      active={false}
      aria-hidden="true"
      aria-label="Ok"
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="close"
      disabled={false}
      onClick={[Function]}
    >
      <Icon
        name="close"
        type="pf"
      />
    </Button>
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Message dialog
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <p />
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      key="btn1"
    >
      Button 1
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      key="btn2"
    >
      Button 2
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      key="btn3"
    >
      Button 3
    </Button>
  </ModalFooter>
</Modal>
`;

exports[`Dialog enables to override cancel button label 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <Button
      active={false}
      aria-hidden="true"
      aria-label="Custom cancel"
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="close"
      disabled={false}
      onClick={[Function]}
    >
      <Icon
        name="close"
        type="pf"
      />
    </Button>
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Message dialog
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <p>
      Some important message
    </p>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="btn-cancel"
      disabled={false}
      key="cancel"
      onClick={[Function]}
    >
      Custom cancel
    </Button>
  </ModalFooter>
</Modal>
`;

exports[`Dialog enables to set inner html 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <Button
      active={false}
      aria-hidden="true"
      aria-label="Ok"
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="close"
      disabled={false}
      onClick={[Function]}
    >
      <Icon
        name="close"
        type="pf"
      />
    </Button>
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Message dialog
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <p
      dangerouslySetInnerHTML={
        Object {
          "_html": "<b>Custom</b> html content",
        }
      }
    />
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="btn-cancel"
      disabled={false}
      key="cancel"
      onClick={[Function]}
    >
      Ok
    </Button>
  </ModalFooter>
</Modal>
`;

exports[`Dialog renders a message dialog 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <Button
      active={false}
      aria-hidden="true"
      aria-label="Ok"
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="close"
      disabled={false}
      onClick={[Function]}
    >
      <Icon
        name="close"
        type="pf"
      />
    </Button>
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Message dialog
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <p>
      Some important message
    </p>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="btn-cancel"
      disabled={false}
      key="cancel"
      onClick={[Function]}
    >
      Ok
    </Button>
  </ModalFooter>
</Modal>
`;
